config/functions

```diff
@@ -217,6 +201,41 @@ strip_gold() {
   LDFLAGS=`echo $LDFLAGS | sed -e "s|-fuse-ld=gold||g"`
 }
 
+# search various project dirs for a given file ($1)
+# if file is not found return a default file   ($2)
+get_project_file() {
+  local d_list=( # main directory roots to search
+    $PROJECT_DIR/$PROJECT/packages
+    $PROJECT_DIR/$PROJECT
+    $HOME/.openelec/projects/$PROJECT/packages
+    $HOME/.openelec/projects/$PROJECT
+    $HOME/.openelec/packages
+    $HOME/.openelec
+  )
+  local s_list=( # subdirectories to search
+    $PKG_NAME
+    .
+  )
+  local result
+  for dir in ${d_list[@]} ; do
+    for sdir in ${s_list[@]} ; do
+      if [ -f $dir/$sdir/$1 ]; then
+        result="$dir/$sdir/$1"
+        break 2
+      fi
+    done
+  done
+  if [ -n "$2" ]; then
```

Comment:
Change to `if [ -z "$result" -a -n "$2" ]; then`

and you don't need to check any further for `-z $result` in lines 229 and 231, and line 231 can be a simple `else` since you know that `$2` won't exist.